apm-agent-core/src/main/java/co/elastic/apm/agent/impl/ElasticApmTracer.java

```diff
@@ -317,6 +321,10 @@ public Span startSpan(AbstractSpan<?> parent, long epochMicros) {
         Transaction transaction = currentTransaction();
         if (transaction != null) {
             if (isTransactionSpanLimitReached(transaction)) {
+                if (lastSpanMaxWarningTimestamp == 0L || (epochMicros - lastSpanMaxWarningTimestamp > MAX_LOG_INTERVAL_MICRO_SECS)) {
```

Comment:
```suggestion
                if (epochMicros - lastSpanMaxWarningTimestamp > MAX_LOG_INTERVAL_MICRO_SECS) {
```
The `lastSpanMaxWarningTimestamp == 0L` check is redundant.